document-generator-api/src/main/java/uk/gov/companieshouse/document/generator/api/service/impl/DocumentGeneratorServiceImpl.java

```diff
@@ -178,4 +176,33 @@ private DocumentResponse setDocumentResponse(RenderDocumentResponse renderRespon
 
         return response;
     }
+
+    /**
+     * Create and log the error message
+     *
+     * @param message The message to be logged
+     * @param <T> generic exception parameter to be stored in message
+     * @param resourceId The resource Id
+     * @param resourceUri The resource Url
+     * @param requestId The request Id
+     */
+    private <T extends Exception> void createAndLogErrorMessage(String message, T exception,
+                                                                String resourceId, String resourceUri, String requestId) {
+
+        Map < String, Object > debugMap = new HashMap < > ();
+        debugMap.put("resource_uri", resourceUri);
+        debugMap.put("resource_id",resourceId);
```

Comment:
Given these keys are used in different classes they could be placed in shared constants or an enum to ensure consistency between all the logs?

Reply:
I have a story to refine the logging as I need to go over the details with yourself.

Reply:
SFA-722